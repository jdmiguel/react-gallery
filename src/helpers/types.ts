export type ThumbData = {
  id: number;
  src: string;
  title: string;
}

export type DetailData = {
  id: number;
  src: string;
  title: string;
  views: number;
  downloads: number;
  favorites: number;
  likes: number;
}

export type ImageExtendedData = {
  id: number;
  pageURL: string;
  type: string;
  tags: string;
  previewURL: string;
  previewWidth: number;
  previewHeight: number;
  webformatURL: string;
  webformatWidth: number;
  webformatHeight: number;
  largeImageURL: string;
  fullHDURL: string;
  imageURL: string;
  imageWidth: number;
  imageHeight: number;
  imageSize: number;
  views: number;
  downloads: number;
  favorites: number;
  likes: number;
  comments: number;
  user_id: number;
  user: string;
  userImageURL: string;
}