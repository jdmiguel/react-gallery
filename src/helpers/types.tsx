export type Image = {
  src: string;
}