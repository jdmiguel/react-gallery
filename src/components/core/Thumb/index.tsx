import React from 'react';
import { Link } from 'react-router-dom';
import styled, { css }  from 'styled-components';
import { LazyImage } from 'react-lazy-images';

import Default from './default.png';

export type ThumbProps = {
  id: number;
  src: string;
  title: string;
}

const ImageCSS = css`
  width: 100%;
  height: 100%;
  max-width: 600px;
  border: 1px solid ${({theme}) => theme.palette.LIGHT_MIN};
`;

const StyledThumb = styled(Link)`
  cursor: pointer;
  width: 100%;
  height: 100%;
  
  > div{
    &:last-of-type{
      width: 100%;
      height: 100%;
    }
  }
  &:hover {
    > div{
      &:first-of-type{
        visibility: visible;
        opacity: 1;
      }
    }
  }
`;

const StyledOverlay = styled.div`
  width: 91%;
  height: 100%;
  position: absolute;
  left: 5%;
  top: 1%;
  z-index: 1;
  background-color: rgba(229, 234, 237, 0.8);
  transition: opacity 0.3s ease-out;
  visibility: hidden;
  opacity: 0;
  transform: scale(0.9);
  display: flex;
  justify-content: center;
  align-items: center;
`;

const StyledThumbText = styled.p`
  font-size: 1.3rem;
  text-transform: uppercase;
  font-weight: 400;
  color: ${({theme}) => theme.palette.PRIMARY_DARK};
`;

const StyledImage = styled(LazyImage)`
  ${ImageCSS}
`;

const StyledDefault = styled.img`
  ${ImageCSS}
`;

const Thumb: React.FC<ThumbProps>  = ({id, src, title}) => (
  <StyledThumb to={`detail/${id}`}>
    <StyledOverlay>
      <StyledThumbText>{title}</StyledThumbText>
    </StyledOverlay>
    <StyledImage
      src={src}
      alt={title}
      placeholder={({ imageProps, ref }) => (
        <div ref={ref} className="LazyImage-Placeholder">
          <StyledDefault
            {...imageProps}
            src={Default}
            alt={title}
          />
        </div>
      )}
      actual={({ imageProps }) => (
        <div className="LazyImage-Actual">
          <img {...imageProps} alt={title} />
        </div>
      )}
    />
  </StyledThumb>
);

export default Thumb;