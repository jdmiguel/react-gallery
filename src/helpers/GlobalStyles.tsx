import { createGlobalStyle } from 'styled-components';

export const GlobalStyles = createGlobalStyle`
  body {
    margin: 0;
    padding: 0;
    font-family: 'Josefin Sans', sans-serif;
    font-size: 15px;
    color: #2f3235;
    background-color: #F3F3F3;
  }

  h1, h2, h3, h4, h5, h6, p{
    margin: 0;
  }

  #root{
    overflow: auto;
  }
`;