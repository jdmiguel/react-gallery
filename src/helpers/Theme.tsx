import React, { ReactNode } from 'react';
import { DefaultTheme, ThemeProvider } from 'styled-components';

export const theme: DefaultTheme = {
  palette: {
    PRIMARY: '#A66885',
    PRIMARY_DARK: '#734359',
    PRIMARY_LIGHT: '#F2C1AE',
    LIGHT_MAX: '#E5EAED',
    LIGHT_MIN: '#C8C8C8',
    GREY: '#E5EAED',
    DARK_MAX: '#2f3235',
    DARK_MIN: '#798186',
  },
};

export const renderWithTheme = (component: ReactNode) => (
  <ThemeProvider theme={theme}>{component}</ThemeProvider>
);
