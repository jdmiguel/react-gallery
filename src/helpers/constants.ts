// API
export const API_PIXABAY = "https://pixabay.com/api/";
export const API_KEY = "6473511-0417f2cad683f1bee54cafe15";

// IMAGES
export const DEFAULT_THUMBS = [
  {
    id: 0,
    src: '',
    title: ''
  },
];

// IMAGE LOADING
export const INITIAL_PAGE = 1;
export const MAX_PAGES_ALLOWED = 20;
export const INITIAL_MAX_IMAGES_LOADED = 40;
export const MAX_IMAGES_LOADED_BY_SCROLL = 20;
export const MIN_FOOTER_VIEWS = 3;
export const OBSERVER_OPTIONS = {
  root: document.querySelector('root'),
  rootMargin: '15%',
  threshold: 0,
};