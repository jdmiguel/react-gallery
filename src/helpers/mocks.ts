export const THUMB_PROPS_MOCKED = {
  id: 3063284,
  src: "https://pixabay.com/get/55e0d340485aa814f1dc84609629327f113dd7e05b4c704f752f78d79744c15e_640.jpg",
  title: "flower"
};
export const BUTTON_TEXT_MOCKED = "default button";
export const Pill_TEXT_MOCKED = "default pill";