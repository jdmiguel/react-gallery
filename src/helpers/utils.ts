export const getLoweredText = (text: string) => text.toLowerCase();