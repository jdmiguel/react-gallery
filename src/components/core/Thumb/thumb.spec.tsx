import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router';

import Thumb from '.';

import { THUMB_PROPS_MOCKED } from '../../../helpers/mocks';
import { renderWithTheme } from '../../../helpers/Theme';

describe('Component: Thumb', () => {
  it('should render', () => {
    const { container } = render(
      renderWithTheme( 
        <MemoryRouter>
          <Thumb {...THUMB_PROPS_MOCKED}/>
        </MemoryRouter>
      )
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
