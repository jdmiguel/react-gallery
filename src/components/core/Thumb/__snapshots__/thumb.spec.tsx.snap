// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Thumb should render 1`] = `
.c0 {
  cursor: pointer;
  width: 100%;
  height: 100%;
}

.c0 > div:last-of-type {
  width: 100%;
  height: 100%;
}

.c0:hover > div:first-of-type {
  visibility: visible;
  opacity: 1;
}

.c1 {
  width: 91%;
  height: 100%;
  position: absolute;
  left: 5%;
  top: 1%;
  z-index: 1;
  background-color: rgba(229,234,237,0.8);
  -webkit-transition: opacity 0.3s ease-out;
  transition: opacity 0.3s ease-out;
  visibility: hidden;
  opacity: 0;
  -webkit-transform: scale(0.9);
  -ms-transform: scale(0.9);
  transform: scale(0.9);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 {
  font-size: 1.3rem;
  text-transform: uppercase;
  font-weight: 400;
  color: #734359;
}

.c4 {
  width: 100%;
  height: 100%;
  max-width: 600px;
  border: 1px solid #C8C8C8;
}

.c3 {
  width: 100%;
  height: 100%;
  max-width: 600px;
  border: 1px solid #C8C8C8;
}

<a
  class="c0"
  href="/detail/3063284"
>
  <div
    class="c1"
  >
    <p
      class="c2"
    >
      flower
    </p>
  </div>
  <div
    class="LazyImage-Placeholder"
  >
    <img
      alt="flower"
      class="c3 c4"
      src="defaultThumb.png"
    />
  </div>
</a>
`;
